lib/debug/thread_client.rb

```diff
@@ -340,11 +340,21 @@ def instance_eval_for_cmethod frame_self, src
       frame_self.instance_eval(src)
     end
 
+    SPECIAL_LOCALS = [
+      [:raised_exception, "_raised_"],
+      [:return_value, "_returned_"],
+    ]
+
     def frame_eval src, re_raise: false
       begin
         @success_last_eval = false
 
         b = current_frame&.eval_binding || TOPLEVEL_BINDING
+        b = b.dup
+
+        SPECIAL_LOCALS.each do |m, local_name|
+          b.local_variable_set(local_name, current_frame.send(m))
```

Comment:
do you need to define lvar if it is not provided?

Reply:
in other tools like `pry`, accessing special variables (e.g. `_ex_`) when they don't exist it just return `nil` instead raising exceptions. I think this would be a nicer experience to have.
also, because `WatchBreakpoint` listens to both `*_return` (has return value) and `call` events, it'll be easier to script bp commands if it doesn't raise exceptions.

```
watch .... do: p _returned_
# instead of 
watch .... do: p _returned_ if: local_variables.include?(:_returned_)
```

Reply:
I don't agree because showing these variables is some kind of signals to show the special lines.
I'll merge this PR and change this points (and variable names).

I'll remove `#return` and use these variables.

```
(rdbg) i    # info command
%self = main
%return = 10
```

Reply:
I'm not sure what the `#return` means here? so just `_raised_` and it'll be renamed as `%raised`?

Reply:
sorry I meant removing `%return`  and show only `_return`